src/transformers/generation/utils.py

```diff
@@ -1794,6 +1795,8 @@ def generate(
                     )
 
                 model_kwargs["past_key_values"] = cache_class(cache_config)
+            elif generation_config.cache_implementation == "offloaded":
+                model_kwargs["past_key_values"] = OffloadedCache()
```

Comment:
why don't we update the mapping that gives us the cahce class here?

Reply:
There is no such mapping? The only mappings I see in the code are
```python
 NEED_SETUP_CACHE_CLASSES_MAPPING = {"static": StaticCache, "sliding_window": SlidingWindowCache, "hybrid": HybridCache}
 QUANT_BACKEND_CLASSES_MAPPING = {"quanto": QuantoQuantizedCache, "HQQ": HQQQuantizedCache}
```
OffloadedCache does not need setup and is not a quantized cache.

Reply:
Yeah I was thinking about the `NEED_SETUP_CACHE_CLASSES_MAPPING`, but alrgith